### PR TITLE
[ARCTIC-1774] Update the scheduling policy when updating optimizer group

### LINKED ISSUE #1774: [Improvement]: Update the scheduling policy without restarting AMS
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/arctic/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Currently, AMS needs to restart when updating the scheduling policy in the optimizer group for not updating the scheduling policy when updating the optimizer group in the `OptimizingQueue`.

### How should we improve?

Update the scheduling policy when needed.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by HuangFru**
## Why are the changes needed?

Fix #1774.

## Brief change log

  - *Add a function to update the table sorter if needed.*
  - *Call the function when updating the optimizer group*

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: ams/server/src/main/java/com/netease/arctic/server/optimizing/SchedulingPolicy.java [hunk 1]
```diff
@@ -23,13 +23,13 @@ public class SchedulingPolicy {
   private static final String BALANCED = "balanced";
 
   private final Map<ServerTableIdentifier, TableRuntime> tableRuntimeMap = new HashMap<>();
-  private final Comparator<TableRuntime> tableSorter;
+  private Comparator<TableRuntime> tableSorter;
   private final Lock tableLock = new ReentrantLock();
 
   public SchedulingPolicy(ResourceGroup group) {
     String schedulingPolicy = Optional.ofNullable(group.getProperties())
```
Comment (zhoujinsong) [inline ams/server/src/main/java/com/netease/arctic/server/optimizing/SchedulingPolicy.java:30]: Recommend making `updateTableSorterIfNeeded` more generic, and we call it directly here.

Comment (HuangFru) [reply ams/server/src/main/java/com/netease/arctic/server/optimizing/SchedulingPolicy.java]: Modified in the new commit.

**Review by zhoujinsong — APPROVED** [review]
LGTM.

File: ams/server/src/main/java/com/netease/arctic/server/optimizing/SchedulingPolicy.java [hunk 2]
```diff
@@ -23,16 +23,22 @@ public class SchedulingPolicy {
...
-    if (StringUtils.isBlank(schedulingPolicy) || schedulingPolicy.equalsIgnoreCase(QUOTA)) {
+    setTableSorterIfNeeded(group);
+  }
+
+  public void setTableSorterIfNeeded(ResourceGroup optimizerGroup) {
+    String schedulingPolicy = Optional.ofNullable(optimizerGroup.getProperties())
+        .orElseGet(Maps::newHashMap)
+        .get(SCHEDULING_POLICY_PROPERTY_NAME);
+    if ((StringUtils.isBlank(schedulingPolicy) || schedulingPolicy.equalsIgnoreCase(QUOTA)) &&
+        tableSorter == null || (tableSorter instanceof BalancedSorter)) {
```
Comment (XBaith) [inline ams/server/src/main/java/com/netease/arctic/server/optimizing/SchedulingPolicy.java:38]: what's the purpose of the condition: `tableSorter instanceof BalancedSorter`?

Comment (zhoujinsong) [reply ams/server/src/main/java/com/netease/arctic/server/optimizing/SchedulingPolicy.java:38]: I assume it covers the situation that `tableSorter` is already initiated and is the same as the new configuration set.

**Pull request merged**